File: data_structures/Arrays/Arrays.php
<?php
/**
 * @author Jose Chavarría
 * @license MIT
 * @link https://chavarria.cr/post/arrays
 */

class Arrays 
{
	function __construct()
	{
		$arrayName = array();
		$this->Insert();
	}

	function Insert(){
		echo("Insert...");
		for ($i=0; $i <5; $i++) { 
			echo '</br>'.'index: '.$i.' value: '.$this->arrayName[]=$i;
		}
	}

	function Traverse(){
		echo("</br></br>Traverse...");
		for ($i=0; $i < count($this->arrayName); $i++) { 
			echo '</br>'.'index: '.$i.' value: '.$this->arrayName[$i];
		}
	}

	function Update($past, $var){
		echo ("</br></br>Update: ".$past." to ".$var);
		for ($i=0; $i < count($this->arrayName); $i++) { 
			if($this->arrayName[$i]==$past){
				echo '</br>'.'index: '.$i.' value: '.$this->arrayName[$i] =$var;
			}
		}
	}
	function Search($var){
		echo ("</br></br>Search :".$var);
		for ($i=0; $i < count($this->arrayName); $i++) { 
			if ($this->arrayName[$i]==$var) {
				echo '</br>'.'index: '.$i.' value: '.$this->arrayName[$i];
			} 
		}
	}
	function Deletion($var){
		echo ("</br></br>Deletion: ".$var."</br>");
		$newArray = array(); $aux=0;
		for ($i=0; $i < count($this->arrayName) ;$i++) { 
			if($this->arrayName[$i]!=$var){
				$this->newArray[$aux]=$this->arrayName[$i];
				$aux++;
			}
		}
		$this->arrayName = $this->newArray;
	}
}

$a = new Arrays();
$a->Traverse();
$a->Search(2);
$a->Update(2,100);
$a->Deletion(3);
$a->Traverse();

?>

File: data_structures/Arrays/Java/Arrays.java
package arrays;

/**
 * @author Jose Chavarría
 * @license MIT
 * @link https://chavarria.cr/post/arrays
 */
public class Arrays {

    /**
     * @param args the command line arguments
     */
    static int[] num = new int[5]; 

    static void Insertion() {
         System.out.println("Insert...");
        for (int i = 0; i < num.length; i++) {
            num[i] = i + 10;
             System.out.println(num[i] = i + 10);
        }
    }

    static void Traverse() {
        System.out.println("\nTraverse...");
        for (int i = 0; i < num.length; i++) {
            System.out.println("index: " + i + " element: " + num[i]);
        }
    }

    static void Search(int var) {
        System.out.println("\nSearch: " + var);
        for (int i = 0; i < num.length; i++) {
            if (num[i] == var) {
                System.out.println("index: " + i + " element: " + num[i]);
            }
        }
    }

    static void Update(int past, int item) {
         System.out.println("\nUpdate: " + past + " to " + item);
        for (int i = 0; i < num.length; i++) {
            if (num[i] == past) {
                num[i] = item;
                 System.out.println("index: " + i + " element: " + num[i]);
            }
        }
    }

    static void Deletion(int var) {
        System.out.println("\nDelete: " + var);
        int[] temp = new int[4];
        int count = 0, aux = 0;
        
        while (count < num.length) {
            if (num[count] != var) {
                temp[aux] = num[count];
                aux++;
            }
            count++;
        }
        num = temp;
    }

    public static void main(String[] args) {
        Insertion();
        Traverse();
        Search(10);
        Update(10, 100);
        Deletion(100);
        Traverse();
    }

}
